src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs

```diff
@@ -253,31 +254,42 @@ internal SymbolTreeInfo Create()
                     try
                     {
                         _metadataReader = moduleMetadata.GetMetadataReader();
+
+                        // First, walk all the symbols from metadata, populating the parentToChilren
+                        // map accordingly.
+                        GenerateMetadataNodes();
+
+                        // Now, once we populated the initial map, go and get all the inheritance 
+                        // information for all the types in the metadata.  This may refer to 
+                        // types that we haven't seen yet.  We'll add those types to the parentToChildren
+                        // map accordingly.
+                        PopulateInheritanceMap();
+
+                        // Clear the set of type definitions we read out of this piece of metadata.
+                        _allTypeDefinitions.Clear();
                     }
-                    catch (BadImageFormatException)
+                    catch (BadImageFormatException ex) when (ReportWatson(ex))
                     {
+                        // any operation off metadata can throw BadImageFormatException
                         continue;
                     }
-
-                    // First, walk all the symbols from metadata, populating the parentToChilren
-                    // map accordingly.
-                    GenerateMetadataNodes();
-
-                    // Now, once we populated the initial map, go and get all the inheritance 
-                    // information for all the types in the metadata.  This may refer to 
-                    // types that we haven't seen yet.  We'll add those types to the parentToChildren
-                    // map accordingly.
-                    PopulateInheritanceMap();
-
-                    // Clear the set of type definitions we read out of this piece of metadata.
-                    _allTypeDefinitions.Clear();
                 }
 
                 var unsortedNodes = GenerateUnsortedNodes();
                 return CreateSymbolTreeInfo(
                     _solution, _checksum, _reference.FilePath, unsortedNodes, _inheritanceMap);
             }
 
+            private bool ReportWatson(BadImageFormatException ex)
+            {
+                // We want to report BadImageFormatException
+                // to make sure it is not due to some bug in metadata reader
```

Comment:
I'm not sure we will get anything useful here. I'd rather we displayed user visible diagnostic and let the user report an issue to us where they can share the offending assembly.

Reply:
@jinujoseph ?

for now, I will remove NFW from it then.